### PR TITLE
[12.0] refactor is_valid_iban

### PR DESCRIPTION
**Pull request by robinkeunen**
- fix failing `test_compute_validated_lines_on_subscription_request` test  but 

... one thing leading to another ...

- more explicit variable name
- test no longer check the id number validity

### CODE REVIEW

**Comment by robinkeunen** [comment]
This branch still fails because of a test in base branch.

**Comment by robinkeunen** [comment]
@carmenbianca @coccinouille CI is now green

File: cooperator/models/subscription_request.py [hunk 1]
```diff
@@ -300,19 +302,13 @@ def _compute_subscription_amount(self):
     )
     user_id = fields.Many2one("res.users", string="Responsible", readonly=True)
     # todo rename to valid_subscription_request
-    validated = fields.Boolean(
-        compute="_compute_validated_lines",
+    is_valid_iban = fields.Boolean(
+        compute="_compute_is_valid_iban",
         string="Valid Subscription request?",
```
Comment (carmenbianca) [inline cooperator/models/subscription_request.py]:
```suggestion
        string="Valid IBAN?",
```

File: cooperator/models/subscription_request.py [hunk 2]
```diff
@@ -300,19 +302,13 @@ def _compute_subscription_amount(self):
...
-    skip_control_ng = fields.Boolean(
-        string="Skip control",
-        help="if this field is checked then no"
-        " control will be done on the national"
-        " register number and on the iban bank"
-        " account. To be done in case of the id"
-        " card is from abroad or in case of"
-        " a passport",
+    skip_iban_control = fields.Boolean(
+        string="Skip control", help="Check to bypass iban format control."
```
Comment (carmenbianca) [inline cooperator/models/subscription_request.py]:
```suggestion
        string="Skip IBAN Control", help="Check to bypass iban format control."
```

File: cooperator/tests/test_emc.py [hunk 3]
```diff
@@ -162,30 +162,26 @@ def test_user_rights(self):
...
-        self.assertFalse(request.iban)
+        request.skip_iban_control = False
 
         # empty iban - don't skip
-        self.assertFalse(request.validated)
+        self.assertFalse(request.is_valid_iban)
 
         # good iban - don't skip
-        request.skip_control_ng = False
         self.request.iban = "BE71096123456769"
```
Comment (carmenbianca) [inline cooperator/tests/test_emc.py:174]: Is this a known unused IBAN?

File: cooperator/tests/test_emc.py [hunk 4]
```diff
@@ -162,30 +162,26 @@ def test_user_rights(self):
...
-        self.assertTrue(request.validated)
+        self.assertTrue(request.is_valid_iban)
 
         # wrong iban - don't skip
         self.request.iban = "xxxx"
-        self.assertFalse(request.validated)
+        self.assertFalse(request.is_valid_iban)
```
Comment (carmenbianca) [inline cooperator/tests/test_emc.py:179]: Missing test for IBAN that has an incorrect check digit.

File: cooperator/views/subscription_request_view.xml [hunk 5]
```diff
@@ -27,13 +27,13 @@
                 <field name="partner_id" />
                 <field name="source" />
                 <field name="state" />
-                <field name="validated" invisible="True" />
+                <field name="is_valid_iban" invisible="True" />
                 <button
                     type="object"
                     string="Validate"
```
Comment (carmenbianca) [inline cooperator/views/subscription_request_view.xml:33]:
```suggestion
                    string="Validate IBAN"
```

**Review by carmenbianca — CHANGES_REQUESTED** [review]
Mostly LGTM, some changes

File: cooperator/tests/test_emc.py (hunk 3 above)
Comment (robinkeunen) [reply cooperator/tests/test_emc.py:174]: I got it from the interweb : https://countrywisecodes.com/belgium/verify-iban-structure/BE71096123456769

File: cooperator/views/subscription_request_view.xml (hunk 5 above)
Comment (robinkeunen) [reply cooperator/views/subscription_request_view.xml:33]: This one is to validate the subscription request and send an invoice to the cooperator candidate.

File: cooperator/tests/test_emc.py (hunk 4 above)
Comment (robinkeunen) [reply cooperator/tests/test_emc.py:179]: I added that 👌

File: cooperator/tests/test_emc.py [hunk 6]
```diff
@@ -162,30 +162,30 @@ def test_user_rights(self):
...
+
+        # wrong iban - don't skip
+        self.request.iban = "BE71096123456760gs"
```
Comment (carmenbianca) [inline cooperator/tests/test_emc.py]:
```suggestion
        self.request.iban = "BE81096123456769"
```

The first two digits in an IBAN are control digits (ECC). They're like a checksum for all the other digits. If we change a control digit, the IBAN verification should fail.

Or is this check more elementary? Does it support Dutch-style IBANs with text in them? e.g. 'NL91ABNA0417164300'? What happens if the IBAN is spaced out à la 'NL91 ABNA 0417 1643 00'?

**Review by carmenbianca — CHANGES_REQUESTED** [review]

File: cooperator/tests/test_emc.py (hunk 6 above)
Comment (robinkeunen) [reply cooperator/tests/test_emc.py]: The `gs` is a typo, I removed it. I intended to fail the checksum by replacing the trailing 9 to 0 and it works. 
The check is actually performed by Odoo Standard so I don't think we should go further in the unit tests.

Comment (carmenbianca) [reply cooperator/tests/test_emc.py]: Ah, got it!

**Review by carmenbianca — APPROVED** [review]

**Pull request merged**